### DISCUSSION TITLE
Request Integration with https://revoke.cash/

### DISCUSSION THREAD

**Discussion by hackfisher** [ringdao-community-guild/discussion#86] [Integration]
https://revoke.cash/

Support Darwinia/Crab and testnets

**Comment by nblogist** [comment]
Hi @hackfisher 

Requirements to list:
- [x] Network listed on [ethereum-lists/chains](https://github.com/ethereum-lists/chains) & [@revoke.cash/chains](https://github.com/RevokeCash/chains)
- [x] [Brand Assets](https://darwinia.notion.site/Brand-Assets-9bd35f88f0674232954737577616f2c2)
- [x] Make a PR with network added

Questions:
1. Does Crab and Darwinia RPC endpoint supports `eth_getLogs`  requests for the entire history of the network? - Where can I get documentation for : https://api.darwinia.network/ ? - or if you'd like me to use subscan in case it have API endpoint, pls lmk.
2. One para description for each Crab and Darwinia CC: @1022ranji #92 
3. Is `multicall3` supported on the networks?
4. What app contracts can I give allowance to on the networks to test revoke.cash with Crab/Darwinia Support

[Source](https://github.com/RevokeCash/revoke.cash)

**Reply by 1022ranji** [reply]
In discussion #92, I updated the introduction to Darwinia. Here is the current introduction for Crab:
Crab is a canary network with real economic value for Darwinia, similar in positioning to Polkadot's Kusama Network. It serves as a programmable cross-chain platform for DApps. Being substrate-based and EVM-compatible, it is connected to Kusama.

**Reply by hackfisher** [reply]
I am including @boundless-forest and @hujw77 to support and answer these questions

**Reply by hujw77** [reply]
For question 3 and 4:
3: Supported, you could check at https://www.multicall3.com
4: You could test on Snowswap Router
  - Darwinia: [0xB899409cDA0fFA2bF87F9c7b31f3c77D6A3a0bB0](https://darwinia.subscan.io/account/0xB899409cDA0fFA2bF87F9c7b31f3c77D6A3a0bB0?tab=transaction)
  - Crab: [0xaf5caa87a7d3718622604268c43ff6ce9d2cec3c](https://crab.subscan.io/account/0xaf5caa87a7d3718622604268c43ff6ce9d2cec3c?tab=contract)

**Reply by boundless-forest** [reply]
> Does Crab and Darwinia RPC endpoint supports eth_getLogs requests for the entire history of the network? - Where can I get documentation for : https://api.darwinia.network/ ? - or if you'd like me to use subscan in case it have API endpoint, pls lmk.

Sure, check out the RPC supported list https://docs.darwinia.network/json-rpc-api-c96e3afedb0b4fb8a902527ffdb5ce6c. The RPC usage is the same as the Ethereum, If there are some inconsistent, note will be added.

**Reply by nblogist** [reply]
@hujw77 Thank for your response


Can you please help me get some clarity on this, I can add a pricing strategy similar to what is there for other chains [here](https://github.com/nblogist/revoke.cash/blob/master/lib/utils/chains.ts#L1094
), Is snowswap a fork of uniswap etc? More importantly is there any stable coin deployed on the networks? Snowswap.xyz isn't displaying any, In which case I'll leave it undefined like the ones [here](https://github.com/nblogist/revoke.cash/blob/master/lib/utils/chains.ts#L1194-L1196) Thank you :)

**Reply by nblogist** [reply]
created a localized PR: https://github.com/nblogist/revoke.cash/pull/1

**Reply by hujw77** [reply]
About stable coin on Darwinia, @jiguantong might know more detail about it.

**Reply by jiguantong** [reply]
Darwina may launch the stablecoin ahUSDT in one month, which is the mapped asset of USDT on AssetHub.

**Reply by nblogist** [reply]
Okay then for now, I'll do what other chains have done [here](https://github.com/nblogist/revoke.cash/blob/master/lib/utils/chains.ts#L1194-L1196) in what I can assume are under similar situation.

**Comment by nblogist** [comment]
Just opened the PR: https://github.com/RevokeCash/revoke.cash/pull/195 Will close this discussion once the PR merges!

**Reply by nblogist** [reply]
Got a request for changes and updated it!

**Comment by nblogist** [comment]
Update: Revoke.cash finally has Darwinia and Crab Network Integrated! https://github.com/RevokeCash/revoke.cash/pull/195
